app/routes/upload.step3.tsx

```diff
@@ -23,22 +26,101 @@ const step3 = () => {
+
+      const imgData = canvas.toDataURL('image/png')
+      const imgWidth = pdf.internal.pageSize.getWidth() // A4 width in mm
+      const imgHeight = (canvas.height * imgWidth) / canvas.width // Maintain aspect ratio
+      const pageHeight = pdf.internal.pageSize.getHeight() // A4 height in mm
+      let heightLeft = imgHeight
+
+      let position = 0
+
+      // Add the first image to the PDF
+      pdf.addImage(imgData, 'PNG', 0, position, imgWidth, imgHeight)
+      heightLeft -= pageHeight
+
+      // If the image height exceeds one page, add more pages
+      while (heightLeft >= 0) {
+        position = heightLeft - imgHeight
+        pdf.addPage()
+        pdf.addImage(imgData, 'PNG', 0, position, imgWidth, imgHeight)
+        heightLeft -= pageHeight
+      }
+
+      // Save the PDF with a specified filename
+      pdf.save('cv.pdf')
+    }
+  }
+
   return (
     <>
       {extractedCV && jobData ? (
         <div>
-          {/* ImproveResume */}
-          {/* <h6>jobTitle</h6>
-          <p className="text-green-600">{jobData.jobTitle}</p>
-          <h6>companyName</h6>
-          <p className="text-green-600">{jobData.companyName}</p>
-          <h6>jobDescription</h6>
-          <p className="text-green-600">{jobData.jobDescription}</p>
-          <h6>CV</h6>
-          <p className="text-green-600">{extractedCV}</p>
-          <hr /> */}
-          {/* <CVDisplay data={enhancedCv} /> */}
-          <CV data={enhancedCv} />
+          <header className="flex justify-between items-center mb-8">
+            <div className="flex items-center space-x-4">
+              {/* <Logo /> */}
+              {/* <div className="flex space-x-2">
+            {[1, 2, 3].map((step) => (
+              <div key={step} className="flex items-center">
+                <div
+                  className={`w-8 h-8 rounded-full border-2 flex items-center justify-center ${step === 3 ? 'border-blue-600 text-blue-600' : 'border-gray-300 text-gray-300'}`}
+                >
+                  {step}
+                </div>
+                {step < 3 && <div className="w-8 h-0.5 bg-gray-300"></div>}
+              </div>
+            ))}
+          </div> */}
+              {/* <Stepper /> */}
+            </div>
+            {/* <div className="flex space-x-2">
+              <Button variant="outline">
+                <Link to="/upload/step1">Adjust to next offer</Link>
+              </Button>
+              <Button onClick={downloadCV}>Download CV</Button>
+            </div> */}
+          </header>
+
+          {/* <main className="grid md:grid-cols-2 gap-8 mb-4">
+            <div>
+              <h2 className="text-xl font-semibold mb-4">Before</h2>
+              <CV data={enhancedCv} />
+            </div>
+            <div>
+              <h2 className="text-xl font-semibold mb-4">After</h2>
+              <div
+                id="cv-to-download"
```

Comment:
**praise**: Great job on implementing the download button with a clear and user-friendly action.